app/controllers/employment.controller.js

```diff
@@ -0,0 +1,59 @@
+const db = require('../models');
+
+const { Employment, User } = db;
+
+// Get all users that are employeed by a user (company)
+exports.userIsEmployedBy = async (req, res, next) => {
+  const {
+    params: { userId },
+  } = req;
+
+  User.findById(userId)
+    .then((user) => user.employedBy())
+    .then((employees) => {
+      return res.status(200).send({
+        message: 'Employees sent successfully',
+        data: employees,
+      });
+    })
+    .catch((err) => next(err));
+};
+
+// Create entry in Employment
+exports.employOneUser = async (req, res, next) => {
+  const {
+    body: { employeeId },
+    params: { userId },
+  } = req;
+
+  User.findById(employeeId)
+    .then(() => {
+      if (employeeId === userId) {
+        throw new StatusError('Employment already exists', 409);
+      }
+    })
+    .then(() => Employment.employ(userId, employeeId))
```

Comment:
this could be simplified. if you add a constraint to the employment migration, you can let the db do the check for you. there's something similar in the promoting table migration file. this way you could just call `Employment.employ` at the start of the chain and have one less query

Reply:
talking about a constraint that doesn't allow employeeId and userId to be equal

Reply:
Thx, done.